src/main/java/org/apache/geronimo/microprofile/impl/jwtauth/jwt/KidMapper.java

```diff
@@ -79,9 +101,48 @@ private void init() {
                                     .collect(Collectors.toSet()))
                                 .orElseGet(HashSet::new);
         ofNullable(config.read("issuer.default", config.read(Names.ISSUER, null))).ifPresent(defaultIssuers::add);
+        jwksUrl = config.read("mp.jwt.verify.publickey.location", null);
+        readerFactory = Json.createReaderFactory(emptyMap());
+        ofNullable(jwksUrl).ifPresent(url -> {
+            HttpClient.Builder builder = HttpClient.newBuilder();
+            if (getJwksRefreshInterval() != null) {
+                long secondsRefresh = getJwksRefreshInterval();
+                backgroundThread = Executors.newSingleThreadScheduledExecutor();
```

Comment:
`ThreadFactory` to name the thread (kill -3 friendly ;))

Reply:
OK, I guess I am not so into the Executor/ThreadFactory stuff. How should this look exactly and why would that help?

Maybe we should put the creation of the Executor in a separate protected method anyway, to allow overriding and using other Executor, wdyt?

Reply:
ok for the protected method for me

for the thread factory just pass a param: `Executors.newSingleThreadScheduledExecutor(worker -> new Thread(worker, KidMapper.class.getName()))` since we'll get a single thread we don't need a counter or anything fancy.

it does not help the code flow but if you do a jstack/kill -3/... on the jvm you will identify  the thread instead of having a banalised `pool-1-thread-1` or anything like that.